### PR TITLE
apps/scan: PDI scanner integration

### PR DESCRIPTION
**Pull request by jonahkagan**
## Overview

Integrates the PDI scanner into VxScan, creating a new state machine that communicates with the scanner using libs/pdi-scanner. Retains support for the Custom scanner by hiding the PDI scanner integration behind a feature flag.

There are a few key differences between this state machine and the Custom scanner state machine:
- PDI scanner scans asynchronously, sending events when a scan starts and completes, so we have to listen for events rather than poll scanner status
- Due to this async mode, we can't easily check once to see if scanning is enabled (e.g. if auth is in voter mode, etc) before starting a scan. Instead, we have to poll to detect when scanning enabled-ness changes and then enable/disable scanning on the scanner accordingly.
- Instead of mapping raw scanner status objects to a few state machine events (e.g. `NO_PAPER` or `READY_TO_SCAN`), I instead opted to let the state machine read the entire status object and used transition guards (`cond`) to make decisions based on the status. This allows different parts of the state machine to interpret the scanner sensors in different ways, rather than needing a universal mapping. (Plus, the old approach was an artifact of the Plustek scanner integration, which only returned a few status strings rather than full sensor data.)

## Demo Video or Screenshot
Basic flows

https://github.com/votingworks/vxsuite/assets/530106/a78f7c1a-e847-425c-be40-cc2783094d7b

Scanning enabled/disabled

https://github.com/votingworks/vxsuite/assets/530106/d920240c-3094-4a24-9914-9acae707189e

Guard before accepting to prevent tailgating ballot

https://github.com/votingworks/vxsuite/assets/530106/44bd3c28-6634-49e3-a76b-844ac756a18e

Jams

https://github.com/votingworks/vxsuite/assets/530106/1373d07f-1988-420d-b223-b2a1e87c7340

## Testing Plan
Manual testing for now. Will be adding automated tests after implementing multi-sheet detection (which is more urgent for our upcoming paper path testing milestone).

## Checklist

- [ ] I have added [logging](https://github.com/votingworks/vxsuite/tree/main/libs/logging) where appropriate to any new user actions, system updates such as file reads or storage writes, or errors introduced.

- [ ] I have added a screenshot and/or video to this PR to demo the change
- [ ] I have added the "user_facing_change" label to this PR to automate an announcement in #machine-product-updates

### CODE REVIEW

**Review by eventualbuddha — APPROVED** [review]
This is looking good. I'm curious what you are thinking for automated testing, as testing the Custom state machine has proven to be flaky. Do you think we can avoid that issue this time around? Is there anything you'd do differently given what you know now?

File: apps/scan/backend/src/scanners/pdi/state_machine.ts [hunk 1]
```diff
@@ -0,0 +1,897 @@
...
+    initial: 'starting',
+    states: {
+      starting: {
+        entry: (context) =>
+          recordRejectedSheet(workspace, usbDrive, context.interpretation),
+        invoke: [
+          {
+            src: async ({ client }) => {
+              (await client.enableScanning()).unsafeUnwrap();
+              (await client.ejectDocument('toFrontAndHold')).unsafeUnwrap();
```
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/scan/backend/src/scanners/pdi/state_machine.ts:226]: We also call it on accept. What do you mean specifically by "in a manner different from this"?

**Comment by jonahkagan** [comment]
> I'm curious what you are thinking for automated testing, as testing the Custom state machine has proven to be flaky.

Great question. I'm not sure yet. One thing I'm hoping will be helpful is the [`SimulatedClock`](https://xstate.js.org/docs/guides/delays.html#interpretation) provided by xstate for testing delays in the state machine. Though not all delays are currently controlled by that clock (specifically the observables), so I'm also considering switching those observables to be child state machines instead so that they can use the same simulated clock. I think controlling time explicitly will remove the flakiness, but there could potentially be other sources of flakiness that I'm not aware of.

File: apps/scan/backend/src/scanners/pdi/state_machine.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (eventualbuddha) [reply apps/scan/backend/src/scanners/pdi/state_machine.ts:226]: i.e. "enable+eject+disable", or effectively the same as that. I see below that you call "eject+disable", but in a state where it must have been enabled. I worry about the tight coupling here but nothing actually enforces that the functions are called in the right order. I can live with it, but I was just looking to add some safety if possible.

**Comment by eventualbuddha** [comment]
> Though not all delays are currently controlled by that clock (specifically the observables), so I'm also considering switching those observables to be child state machines instead so that they can use the same simulated clock.

This sounds promising.

File: apps/scan/backend/src/scanners/pdi/state_machine.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (jonahkagan) [reply apps/scan/backend/src/scanners/pdi/state_machine.ts:226]: Gotcha. Yeah, I see where your coming from. I can't think of any major downsides to abstracting the enabling/disabling of the feeder, so I'm game to try that.

**Pull request merged**